tools/bouncer_worker/src/lib/processMonitor.js

```diff
@@ -1,59 +1,139 @@
 const pidusage = require('pidusage');
 const processExists = require('process-exists');
+const si = require('systeminformation');
 const fs = require('fs');
 const logger = require('./logger');
-const { elastic } = require('./config').config;
+const Elastic = require('./elastic');
+const { enabled, memoryIntervalMS } = require('./config').config.processMonitoring;
+const elasticEnabled = require('./config').config.elastic.enabled;
 
-const processMonitor = {};
-processMonitor.logLabel = { label: 'PROCESSMON' };
+const ProcessMonitor = {};
+const logLabel = { label: 'PROCESSMONITOR' };
 
-const pidArray = [];
-let stats = [];
-let maxMemory = 0;
-let startMemory = 0;
+const informationDict = {};
+const workingDict = {};
+const pidSet = new Set();
+const permittedOS = ['linux', 'win32'];
 
-processMonitor.maxmem = async () => {
+const getCurrentOperatingSystem = async () => {
+	let data;
 	try {
-		const data = Number(fs.readFileSync('/sys/fs/cgroup/memory/memory.usage_in_bytes'));
-		if (pidArray.length > 0) {
-			stats = await pidusage(pidArray);
-			if (data > maxMemory) maxMemory = data;
+		data = await si.osInfo();
+	} catch (e) {
+		logger.error('Failed to get operating system information record', logLabel);
+	}
+	return data.platform;
+};
+const currentOSPromise = getCurrentOperatingSystem();
+
+const getDockerEnvironment = async () => {
+	const currentOS = await currentOSPromise;
+	if (currentOS === 'linux' && fs.existsSync('/.dockerenv')) {
+		return true;
+	}
+	return false;
+};
+const isDockerEnvironment = getDockerEnvironment();
+
+const getCurrentMemUsage = async () => {
+	let data;
```

Comment:
data is not used outside of the try {}, infact it's never overwritten so it should just be a `const`

But with the suggested change you don't even need a variable.